internal/kokoro/continuous.sh

```diff
@@ -53,23 +54,88 @@ try3 go mod download
 go install github.com/jstemmer/go-junit-report
 ./internal/kokoro/vet.sh
 
+# Continuous jobs only run root tests & tests in submodules changed by the PR
+# Nightly jobs run all tests in all submodules
+SIGNIFICANT_CHANGES=$(git --no-pager diff --name-only master..HEAD | grep -Ev '(\.md$|^\.github)' || true)
+# CHANGED_DIRS is the list of significant top-level directories that changed,
+# but weren't deleted by the current PR.
+# CHANGED_DIRS will be empty when run on master.
+CHANGED_DIRS=$(echo "$SIGNIFICANT_CHANGES" | tr ' ' '\n' | grep "/" | cut -d/ -f1 | sort -u | tr '\n' ' ' | xargs ls -d 2>/dev/null || true)
+
+# List all modules in changed directories.
+# If running on master will collect all modules in the repo, including the root module.
+# shellcheck disable=SC2086
+GO_CHANGED_MODULES="$(find ${CHANGED_DIRS:-.} -name go.mod)"
+# If we didn't find any modules, use the root module.
+GO_CHANGED_MODULES=${GO_CHANGED_MODULES:-./go.mod}
+# Exclude the root module, if present, from the list of sub-modules.
+GO_CHANGED_SUBMODULES=${GO_CHANGED_MODULES#./go.mod}
```

Comment:
This does not look like it is used?

Reply:
I rewrote the script and scrapped this part